File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxP = 0
        p1 = 0
        p2 = 1
        while p2 < len(prices):
            if prices[p2] < prices[p1]:
                p1 = p2
                p2 = p1 + 1
            else:
                maxP = max(prices[p2] - prices[p1], maxP)
                p2 += 1
        
        return maxP

File: Data Structures & Algorithms/buy-and-sell-crypto-with-cooldown/submission-1.cpp
class Solution {
public:
    int sol(int i, vector<int>& prices) {
        if (i >= prices.size() - 1) {
            return 0;
        }

        auto ret{0};
        for (auto j{i}; j < prices.size() - 1; ++j) {
            auto max{0};
            for (auto k{j}; k < prices.size(); ++k) {
                if (prices[k] <= prices[j]) {
                    continue;
                }
                max = std::max(max, prices[k] - prices[j] + sol(k + 2, prices));
            }
            ret = std::max(ret, max);
        }
        return ret;
    }
    int maxProfit(vector<int>& prices) {
        return sol(0, prices);
    }
};
